asllib/TopoSort.mli

```diff
@@ -0,0 +1,81 @@
+(** This modules implement a topological sorting algorithm. *)
+
+(** {1 Topological sorting of ASTs.} *)
+
+(** In the following, {i dependencies} are understood in the ASL declaration
+    level. For example, in the following ASL code:
+    {v
+       constant x: integer = 4;
+       constant y: integer = x + 1;
+    v}
+    the declaration of [y] should happen after the declaration of [x],
+    because the declaration of [y] uses the value of [x].
+*)
```

Comment:
The example may lead to believe that, generally speaking ASL definition must appear in source  according to their dependencies, which does not look to be the case. Said otherwise "should happen" above is ambiguous.

Perhaps "should happen" refers to typing or evaluation order?

Besides the question of mutual dependencies is overlooked. Are mutually recursive functions possible?